### PR TITLE
feat: custom shortcuts, fixed reset, optimised 

### PR DESCRIPTION
**Pull request by krakenftw**
## Description

Please provide a clear description of your changes.

Added custom shortcuts, user can record keys and it will be saved in db, pull from db on page load and if overrides found, it will use those as shortcut.

reopened #1438 

---

## Type of Change

Please delete options that are not relevant.

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature with breaking changes)
- [ ] 📝 Documentation update
- [ ] 🎨 UI/UX improvement
- [ ] 🔒 Security enhancement
- [x] ⚡ Performance improvement

## Areas Affected

Please check all that apply:

- [ ] Email Integration (Gmail, IMAP, etc.)
- [ ] User Interface/Experience
- [ ] Authentication/Authorization
- [ ] Data Storage/Management
- [ ] API Endpoints
- [ ] Documentation
- [ ] Testing Infrastructure
- [ ] Development Workflow
- [ ] Deployment/Infrastructure

## Testing Done

Describe the tests you've done:

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [ ] Cross-browser testing (if UI changes)
- [ ] Mobile responsiveness verified (if UI changes)

## Security Considerations

For changes involving data or authentication:

- [ ] No sensitive data is exposed
- [ ] Authentication checks are in place
- [ ] Input validation is implemented
- [ ] Rate limiting is considered (if applicable)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/Mail-0/Zero/blob/staging/.github/CONTRIBUTING.md) document
- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in complex areas
- [ ] I have updated the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix/feature works
- [ ] All tests pass locally
- [ ] Any dependent changes are merged and published

## Additional Notes

Add any other context about the pull request here.

## Screenshots/Recordings

Add screenshots or recordings here if applicable.

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the project's license._

## Summary by CodeRabbit

- **New Features**
  - Enabled editing and resetting of keyboard shortcuts in settings, including hotkey recording and success notifications.
- **Improvements**
  - Keyboard shortcuts are now managed dynamically, allowing real-time updates and resets.
  - Shortcut changes are immediately reflected across navigation, mail list, thread display, and compose views.
- **Bug Fixes**
  - Improved reliability of shortcut updates and resets, with better error handling and user feedback.

### CODE REVIEW

**Comment by ahmetskilinc** [comment]
can you fix the coderabbit comments and then I'll test this asap.

**Comment by krakenftw** [comment]
@ahmetskilinc, there is an edge case i just found, we need to disable hotkey provider when hotkey listener ( for updating shortcut ) is on screen. 

This will cause shortcuts to trigger when recording keys.

Also will need to handle conflicts ( different pr ig )

File: apps/server/src/main.ts [hunk 1]
```diff
@@ -180,10 +181,12 @@ class ZeroDB extends DurableObject {
...
-  async findUserHotkeys(userId: string): Promise<(typeof userHotkeys.$inferSelect)[]> {
-    return await this.db.query.userHotkeys.findMany({
+  async findUserHotkeys(
+    userId: string,
+  ): Promise<(typeof userHotkeys.$inferSelect & { shortcuts: Shortcut[] }) | undefined> {
+    return (await this.db.query.userHotkeys.findFirst({
       where: eq(userHotkeys.userId, userId),
-    });
+    })) as (typeof userHotkeys.$inferSelect & { shortcuts: Shortcut[] }) | undefined;
   }
```
[the remark this answers is not in the document]
Comment (krakenftw) [reply apps/server/src/main.ts]: resolved

**Comment by krakenftw** [comment]
@ahmetskilinc this is what i did on this PR, i think you checked the old PR. this contains 3 commits which seems correct to me. is there anything i am missing?
